devito/grid.py

```diff
@@ -0,0 +1,50 @@
+from devito.tools import as_tuple
+from devito.dimension import x, y, z
+
+import numpy as np
+
+__all__ = ['Grid']
+
+
+class Grid(object):
+
+    """
+    A cartesian grid that encapsulates a physical domain over which
+    to discretize :class:`Function`s.
+
+    :param shape: Shape of the domain region in grid points.
```

Comment:
Is it worth start mentioning the distinction between domain and halo? or is it too premature?

Reply:
I actually tried that, but flipping the automated treatment of the stencil boundary is tricky, so I left if for another PR. So the documentation of these concepts (stencil boundary, halo, etc.) should come with the actual features, I think.